File: app/src/main/java/network/devandroid/com/networklibrarycomparison/okhttp/OkHttpNetworkManager.java
package network.devandroid.com.networklibrarycomparison.okhttp;

import android.os.Handler;
import android.os.Looper;

import java.io.IOException;

import network.devandroid.com.networklibrarycomparison.internal.BaseNetworkManager;
import okhttp3.CacheControl;
import okhttp3.Call;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class OkHttpNetworkManager extends BaseNetworkManager<String> {

    public OkHttpNetworkManager(Callback mCallback) {
        super(mCallback);
    }

    @Override
    public void send(String url) {
        OkHttpClient client = new OkHttpClient.Builder()
                .cache(null)//disable cache
                .build();
        Request request = new Request.Builder()
                .url(url)
                .cacheControl(new CacheControl.Builder().noCache().build())
                .build();
        client.newCall(request).enqueue(new okhttp3.Callback() {
            @Override
            public void onFailure(Call call, final IOException e) {
                if (null != mCallback) {
                    new Handler(Looper.getMainLooper()).post(new Runnable() {
                        @Override
                        public void run() {
                            mCallback.onError(e.getLocalizedMessage());
                        }
                    });
                }
            }

            @Override
            public void onResponse(Call call, final Response response) throws IOException {
                if (null != mCallback) {
                    final String result = response.body().string().toString();
                    new Handler(Looper.getMainLooper()).post(new Runnable() {
                        @Override
                        public void run() {
                            mCallback.onResponse(result);
                        }
                    });
                }
            }
        });
    }


}

File: app/src/main/java/network/devandroid/com/networklibrarycomparison/okhttp/OkHttpNetworkManager.kt
package network.devandroid.com.networklibrarycomparison.okhttp

import android.os.Handler
import android.os.Looper

import java.io.IOException

import network.devandroid.com.networklibrarycomparison.internal.BaseNetworkManager
import network.devandroid.com.networklibrarycomparison.internal.INetworkManager
import okhttp3.CacheControl
import okhttp3.Call
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response

class OkHttpNetworkManager(mCallback: INetworkManager.Callback) : BaseNetworkManager(mCallback) {

    override fun send(url: String) {
        val client = OkHttpClient.Builder()
                .cache(null)//disable cache
                .build()
        val request = Request.Builder()
                .url(url)
                .cacheControl(CacheControl.Builder().noCache().build())
                .build()
        client.newCall(request).enqueue(object : okhttp3.Callback {
            override fun onFailure(call: Call, e: IOException) {
                if (null != callback) {
                    Handler(Looper.getMainLooper()).post { callback.onError(e.localizedMessage) }
                }
            }

            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                if (null != callback) {
                    val result = response.body()!!.string().toString()
                    Handler(Looper.getMainLooper()).post { callback.onResponse(result) }
                }
            }
        })
    }


}
